GetIntrinsic.js

```diff
@@ -255,24 +255,58 @@ module.exports = function GetIntrinsic(name, allowMissing) {
 
 		if (hasOwn(INTRINSICS, intrinsicRealName)) {
 			value = INTRINSICS[intrinsicRealName];
+			desc = { value: value };
 		} else if (value != null) {
 			if ($gOPD && (i + 1) >= parts.length) {
-				var desc = $gOPD(value, part);
+				desc = $gOPD(value, part);
 				isOwn = !!desc;
+				var isMissing = !(part in value);
 
-				if (!allowMissing && !(part in value)) {
+				if (!allowMissing && isMissing) {
 					throw new $TypeError('base intrinsic for ' + name + ' exists, but the property is not available.');
 				}
-				value = isOwn ? desc.get || desc.value : value[part];
+
+				// create a fake prop desc if from a deeper prototype chain
+				desc = desc || { value: value[part] };
+
+				// update the value for the cache
+				value = desc.value || desc.get;
 			} else {
 				isOwn = hasOwn(value, part);
 				value = value[part];
+				// create a fake prop desc
+				desc = { value: value };
 			}
 
 			if (isOwn && !skipFurtherCaching) {
 				INTRINSICS[intrinsicRealName] = value;
 			}
 		}
 	}
-	return value;
+
+	return desc;
```

Comment:
given that the enumerable/configurable/writable properties won't be accurate in a pre-descriptor engine, i'm not sure if it's actually useful to expose this function. maybe it's just a helper for the two other functions?